### ISSUE TITLE
Integrate Cloudinary to Netlify CMS so admins can upload image directly from CMS

### ISSUE THREAD

**Issue by ekafyi** [kawalcovid19/wargabantuwarga.com#395]
Netlify CMS has a [Cloudinary integration](https://www.netlifycms.org/docs/cloudinary/) for the Media assets. Making use of this integration will improve ease of use for admins who need to update WBW images, as they can do everything in one place (WBW admin page)—no need to login to Cloudinary, upload image there, copy the path, etc.

It might be particularly useful for #321 when we have multiple static images.

For further investigation and discussion:
- Media library extension? https://www.netlifycms.org/blog/2019/07/netlify-cms-gatsby-plugin-4-0-0#using-media-libraries-with-netlify-cms-app
- Is there a way to limit uploads & transformations so we don't go overboard with the Cloudinary quota? Or shall we simply inform it verbally in the CMS field description/hint, eg. "No need to edit/transform the images here"?
- Currently the image dimensions (w&h) are hardcoded in the Image component. This is required for built-in next/image lazy-loading. We can either (a) programmatically get image dimensions data, (b) "lock"/predefine the aspect ratio for each banner, or (c) stay with the current setup, ie. hardcode the dimensions in the code.

**Comment by resir014** [comment]
@ekafyi We should probably expedite this issue with #515 now in progress. We should allow content writers to easily upload images to our Cloudinary account.

How long do you think this can be implemented? (cc: @zainfathoni)

**Comment by ekafyi** [comment]
An <abbr title="minimum viable product">MVP</abbr> implementation should be quick to implement. I can finish it by tomorrow. Then we can improve/iterate from there. @resir014 @zainfathoni

**Comment by ekafyi** [comment]
Need help deciding integration option.

## (1) In Markdown widget WYSIWYG editor

### Don't apply any customization
Not recommended, it will return original uncompressed image

### Option A: Let users apply transformation manually from the CMS

![Screen Shot 2021-08-04 at 09 48 32](https://user-images.githubusercontent.com/6597211/128120620-08c0597a-1329-41e2-b8e6-bc8eff9b3e5d.png)
![Screen Shot 2021-08-04 at 09 49 15](https://user-images.githubusercontent.com/6597211/128120634-683ff07e-84e7-411c-a97b-fdb7d6c41c59.png)
![Screen Shot 2021-08-04 at 09 49 53](https://user-images.githubusercontent.com/6597211/128120641-a7c8bddd-9d97-4f32-9039-a43b6262b6cd.png)

Pros
- Maximum versatility. Can have image in Post A sized 900px, image in Post B sized 400px and converted to grayscale, image in Post C sized 120px and hard-cropped to square, etc.

Cons
- Poor UX. Users (content editors/admins) have extra step to do for every image, must remember or be supplied with guidelines on transformation configuration (size, quality etc).
- Users may try arbitrary/unnecessary transformations, wasting our Cloudinary credits.

### Option B: Apply default transformation configuration

We can set reasonable default in `admin/config.yml`, eg. resize proportionally to 640px width, automatic high compression.

```yaml
media_library:
  name: cloudinary
  config:
    # ...
    default_transformations:
      - - crop: scale
          width: 640
          quality: auto:eco
```

Pro:
- Better UX for users, fewer step to take

Con:
- Less versatile than Option A

### Option C: Add custom transformer

Add to `lib/htmr-transformer`:

```js
const img = (node: JSX.IntrinsicElements["img"]) => {
  return ...
};
```

Pros:
- (same as Option B)
- Can apply default custom transformation only for Markdown widget, does not affect image widget

Con:
- (same as Option B)


## (2) In image widget

(in progress)

**Comment by ekafyi** [comment]
Update: Going with Option C (details above) for now. We can always change if/as needed. :)

**Comment by resir014** [comment]
Yep, let's go with C. Let's use as much code-level transformation as possible to make it easier for our content writers. @ekafyi

**Comment by ekafyi** [comment]
It's done @resir014 @mazipan , I just have no idea how to test this part 😅

https://github.com/kawalcovid19/wargabantuwarga.com/blob/6c7577d16107c1ef0cc1ef5b81d9ce86ae34e6ae/lib/htmr-transformers.tsx#L87-L106

Can't use `render` from `@testing-library/react` on the `img` function nor the whole `<div>{htmr(<h1>hello world</h1>, { transform: htmrTransform })}</div>`

**Comment by zainfathoni** [comment]
Let me try to help you with the testing parts tonight, @ekafyi, since I'm the 🧪  test toaster 🍞. 😆

**Comment by ekafyi** [comment]
> Let me try to help you with the testing parts tonight, @ekafyi, since I'm the 🧪 test toaster 🍞. 😆

yay, awesome, thanks @zainfathoni !

I still need to refactor this a little though, since I just found out the `Image` component is not suitable for images inside the Markdown body. If it still fails when I deploy, I'll @ you for help  🍞..

**Comment by ekafyi** [comment]
Welp, I enabled "auto-merge" and it merged to main on check completion because it had an existing approval which I was not aware of. 😬

Everything seems ok as far as I could check—feel free to recheck, anyone.

We've got ✨ proper image integration ✨ in the CMS now!

![Screen Shot 2021-08-06 at 00 48 01](https://user-images.githubusercontent.com/6597211/128397408-c3436588-8b54-4921-ba1b-e1d23070e303.png)

As a heads-up I had to log out then log back in to the Netlify CMS to get rid of old CMS cache (plaintext widget for the Home Banner).
